tests/st/calicoctl/test_crud.py

```diff
@@ -35,745 +35,1091 @@ class TestPool(TestBase):
+        rc.assert_data(ippool_name1_rev1_v4)
+        rc = calicoctl("get ippool -o yaml")
+        rc.assert_list("IPPool", [ippool_name1_rev1_v4, ippool_name2_rev1_v6])
 
-        ipv6_pool_dict = {'apiVersion': 'projectcalico.org/v2',
-                          'kind': 'IPPool',
-                          'metadata': { 'name': 'ippool-456'},
-                          'spec': {'cidr': str(ipv6_net.cidr)}
-                          }
+        # Remove both the ipv4 pool and ipv6 pool by CLI options and by file.
+        rc = calicoctl("delete ippool %s" % name(ippool_name1_rev1_v4))
+        rc.assert_no_error()
+        rc = calicoctl("delete", ippool_name2_rev1_v6)
+        rc.assert_no_error()
 
-        # Write out some yaml files to load in through calicoctl-go
-        # We could have sent these via stdout into calicoctl, but this
-        # seemed easier.
-        self.writeyaml('/tmp/ipv4.yaml', ipv4_pool_dict)
-        self.writeyaml('/tmp/ipv6.yaml', ipv6_pool_dict)
+        # Assert pools are now deleted
+        rc = calicoctl("get ippool -o yaml")
+        rc.assert_empty_list("IPPool")
 
-        # Create the ipv6 network using calicoctl
-        calicoctl("create", "/tmp/ipv6.yaml")
-        # Now read it out (yaml format) with calicoctl:
-        self.check_data_in_datastore([ipv6_pool_dict], "ipPool")
+        # Assert that deleting the pool again fails.
+        rc = calicoctl("delete ippool %s" % name(ippool_name2_rev1_v6))
+        rc.assert_error(text=NOT_FOUND)
 
-        # Add in the ipv4 network with calicoctl
-        calicoctl("create", "/tmp/ipv4.yaml")
-        # Now read it out with the calicoctl:
-        self.check_data_in_datastore([ipv4_pool_dict, ipv6_pool_dict], "ipPool")
+    def test_delete_with_resource_version(self):
+        """
+        Test that resource version operates correctly with delete, i.e.
+        calicoctl honors the resource version when it's specified.
+        """
 
-        # Remove both the ipv4 pool and ipv6 pool
-        calicoctl("delete", "/tmp/ipv6.yaml")
-        calicoctl("delete", "/tmp/ipv4.yaml")
-        # Assert output contains neither network
-        self.check_data_in_datastore([], "ipPool")
+        # Create a new BGP Peer and get it to determine the current resource
+        # version.
+        rc = calicoctl("create", data=bgppeer_name1_rev1_v4)
+        rc.assert_no_error()
+        rc = calicoctl("get bgppeer %s -o yaml" % name(bgppeer_name1_rev1_v4))
+        rc.assert_no_error()
+        rev0 = rc.decoded
 
-        # Assert that deleting the pool again fails.
-        self.assertRaises(CommandExecError, calicoctl, "delete", "/tmp/ipv4.yaml")
+        # Update the BGP Peer and get it to assert the resource version is not
+        # the same.
+        rc = calicoctl("apply", data=bgppeer_name1_rev2_v4)
+        rc.assert_no_error()
+        rc = calicoctl("get bgppeer %s -o yaml" % name(bgppeer_name1_rev1_v4))
```

Comment:
I think this seems a bit confusing that you're doing a get with `rev1` data. I know Get only cares about metadata which hasn't changed but getting with `rev2` would be clearer